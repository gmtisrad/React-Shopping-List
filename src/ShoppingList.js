import React from 'react';
import ShoppingListItem from './ShoppingListItem.js';
import './ShoppingList.css';

class ShoppingList extends React.Component {
    static defaultProps = {
        shoppingList: [],
    }
    renderListItems() {
        return (
            this.props.shoppingList.map((listItem, index) => (
                <ShoppingListItem 
                    key={index} 
                    name={listItem.name} 
                    checked={listItem.checked}
                    onDeleteItem={this.props.onDeleteItem}
                    onCheckItem={this.props.onCheckItem} />
            ))
        )
    }

    render () {
        return (
            <ul className='shopping-list'>
                {this.renderListItems()}
            </ul>
        )
    }
}

export default ShoppingList;