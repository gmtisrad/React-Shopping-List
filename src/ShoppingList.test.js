import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ShoppingList from './ShoppingList.js';

describe('ShoppingList component tests', () => {      
    it('renders empty given no props', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ShoppingList />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders UI as expected with no props', () => {
        const wrapper = shallow(<ShoppingList />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('renders UI as expected with props', () => {
        const shoppingItems = [
            { name: 'apples', checked: false },
        ]
        const wrapper = shallow(<ShoppingList shoppingList={shoppingItems} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('renders UI as expected with multiple props', () => {
        const shoppingItems = [
            { name: 'apples', checked: false },
            { name: 'apples', checked: true },
            { name: 'apples', checked: false },
        ]
        const wrapper = shallow(<ShoppingList shoppingList={shoppingItems} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})