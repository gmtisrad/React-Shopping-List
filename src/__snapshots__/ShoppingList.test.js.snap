// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingList component tests renders UI as expected with multiple props 1`] = `
<ul
  className="shopping-list"
>
  <ShoppingListItem
    checked={false}
    key="0"
    name="apples"
  />
  <ShoppingListItem
    checked={true}
    key="1"
    name="apples"
  />
  <ShoppingListItem
    checked={false}
    key="2"
    name="apples"
  />
</ul>
`;

exports[`ShoppingList component tests renders UI as expected with no props 1`] = `
<ul
  className="shopping-list"
/>
`;

exports[`ShoppingList component tests renders UI as expected with props 1`] = `
<ul
  className="shopping-list"
>
  <ShoppingListItem
    checked={false}
    key="0"
    name="apples"
  />
</ul>
`;
